wagtail/admin/views/pages/listing.py

```diff
@@ -307,32 +266,133 @@ def order_queryset(self, queryset):
         return queryset
 
     def search_queryset(self, queryset):
-        # allow hooks to modify queryset. This should happen as close as possible to the
-        # final queryset, but (for backward compatibility) needs to be passed an actual queryset
-        # rather than a search result object
-        for hook in hooks.get_hooks("construct_explorer_page_queryset"):
-            queryset = hook(self.parent_page, queryset, self.request)
-
         if self.is_searching:
             queryset = queryset.autocomplete(
                 self.query_string, order_by_relevance=(not self.is_explicitly_ordered)
             )
 
         return queryset
 
-    def get_paginate_by(self, queryset):
-        if self.ordering == "ord":
-            # Don't paginate if sorting by page order - all pages must be shown to
-            # allow drag-and-drop reordering
-            return None
+    def get_index_url(self):
+        return reverse(self.index_url_name)
+
+    def get_index_results_url(self):
+        return reverse(self.index_results_url_name)
+
+    def get_breadcrumbs_items(self):
+        return self.breadcrumbs_items + [{"url": "", "label": self.get_page_title()}]
+
+    def get_table_kwargs(self):
+        kwargs = super().get_table_kwargs()
+        kwargs["actions_next_url"] = self.index_url
+        return kwargs
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        context.update(
+            {
+                "ordering": self.ordering,
+                "search_form": self.search_form,
+                "is_searching": self.is_searching,
+            }
+        )
+        return context
+
+
+class ExplorableIndexView(IndexView):
+    """
+    A version of the page listing where the user is presented with a view of a specified parent page;
+    normally this will show the children of that page, but it may show results from the whole tree while
+    searching or filtering.
+    """
+
+    template_name = "wagtailadmin/pages/explorable_index.html"
+    index_url_name = "wagtailadmin_explore"
+    index_results_url_name = "wagtailadmin_explore_results"
+    page_title = _("Exploring")
+    filterset_class = ExplorablePageFilterSet
+
+    columns = (
+        IndexView.columns[0:3]
+        + [
+            Column(
+                "type",
+                label=_("Type"),
+                accessor="page_type_display_name",
+                sort_key="content_type",
+                width="12%",
+            ),
+        ]
+        + IndexView.columns[3:]
+        + [
+            NavigateToChildrenColumn("navigate", width="10%"),
+        ]
+    )
```

Comment:
Would it be better to use Django's [`@classproperty`](https://docs.djangoproject.com/en/stable/ref/utils/#django.utils.functional.classproperty) here? e.g.

```suggestion
    @classproperty
    def columns(cls):
        columns = IndexView.columns.copy()
        columns.insert(
            3,
            Column(
                "type",
                label=_("Type"),
                accessor="page_type_display_name",
                sort_key="content_type",
                width="12%",
            ),
        )
        columns.append(NavigateToChildrenColumn("navigate", width="10%"))
        return columns
```

Reply:
Good idea! Applied in 4340e216e1e94c2ac12a3a5445d3195772adb95d.